iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp

```diff
@@ -316,56 +319,68 @@ class AdjustDepthwiseFilterShape : public OpRewritePattern<mhlo::ConvOp> {
 // into
 //
 // %z = "BinaryElementwiseOpT"(%x, %y) : (%[[SHAPE_BEFORE_BCAST]], %[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_BEFORE_BCAST]]
-// %result = "mhlo.broadcast_in_dim"(%z) {} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
+// %result = "mhlo.broadcast_in_dim"(%z) {broadcast_dimensions = %[[BCAST_DIMS]]} : (%[[SHAPE_BEFORE_BCAST]]) -> %[[SHAPE_AFTER_BCAST]]
 //
 // clang-format on
-template <typename BinaryElementwiseOpT>
-class ReorderBroadcastInDimOpAndBinaryElementwiseOp
-    : public OpRewritePattern<BinaryElementwiseOpT> {
+template <typename ElementwiseOpT>
+class ReorderBroadcastInDimOpAndElementwiseOp
+    : public OpRewritePattern<ElementwiseOpT> {
  public:
-  using OpRewritePattern<BinaryElementwiseOpT>::OpRewritePattern;
+  using OpRewritePattern<ElementwiseOpT>::OpRewritePattern;
 
-  LogicalResult matchAndRewrite(BinaryElementwiseOpT op,
+  LogicalResult matchAndRewrite(ElementwiseOpT op,
                                 PatternRewriter &rewriter) const override {
-    auto lhsType = op.lhs().getType().template dyn_cast<ShapedType>();
-    auto rhsType = op.rhs().getType().template dyn_cast<ShapedType>();
-    if (!lhsType || !rhsType || !lhsType.hasStaticShape() ||
-        lhsType != rhsType) {
-      return failure();
-    }
-
-    auto lhsOp = op.lhs().getDefiningOp();
-    auto rhsOp = op.rhs().getDefiningOp();
-    if (!lhsOp || !rhsOp) {
-      return failure();
+    Operation *operation = op.getOperation();
+    assert(operation->getNumOperands() >= 1 && operation->getNumResults() == 1);
+
+    // Verify if all operands are from BroadcastInDimOp and its
+    // broadcast_dimensions is the same.
+    llvm::SmallVector<mhlo::BroadcastInDimOp, 2> bcastOps;
+    for (auto operand : operation->getOperands()) {
+      auto bcastOp = operand.getDefiningOp<mhlo::BroadcastInDimOp>();
+      if (!bcastOp) {
+        return failure();
+      }
+      bcastOps.push_back(bcastOp);
```

Comment:
[optional] This is just my preference. :) How about:

```
if (auto bcastOp = operand.getDefiningOp<mhlo::BroadcastInDimOp>()) {
  bcastOps.push_back(bcastOp);
} else {
  return failure();
}
```

Reply:
Looks good to me! Thank you for pointing out this!